src/System.Private.CoreLib/src/System/Threading/ThreadPool.cs

```diff
@@ -415,16 +415,27 @@ internal void EnsureThreadRequested()
             // Note that there is a separate count in the VM which will also be incremented in this case, 
             // which is handled by RequestWorkerThread.
             //
+            int procCount = ThreadPoolGlobals.processorCount;
             int count = numOutstandingThreadRequests;
-            while (count < ThreadPoolGlobals.processorCount)
+            if (count < procCount)
             {
-                int prev = Interlocked.CompareExchange(ref numOutstandingThreadRequests, count + 1, count);
-                if (prev == count)
+                do
                 {
-                    ThreadPool.RequestWorkerThread();
-                    break;
-                }
-                count = prev;
+                    int prev = Interlocked.CompareExchange(ref numOutstandingThreadRequests, count + 1, count);
+                    if (prev == count)
+                    {
+                        ThreadPool.RequestWorkerThread();
+                        break;
+                    }
+                    else if (prev >= procCount)
+                    {
+                        break;
+                    }
+
+                    count = prev;
+                    // Missed update, indicate to CPU we are in a spin loop
+                    Thread.SpinWait(1);
```

Comment:
SpinWait or PAUSE is useful when the other thread has to make a progress before it makes sense to retry. It is not the case for the compare exchange loops like this. It should be fine to retry right away with no harm.

Reply:
Ah, k